src/MoBi.Core/Domain/Model/MoBiSimulation.cs

```diff
@@ -131,13 +132,22 @@ public override void UpdatePropertiesFrom(IUpdatable source, ICloneManager clone
 
       public void RemoveUsedObservedData(DataRepository dataRepository)
       {
-         
+         if (!UsesObservedData(dataRepository))
+            return;
 
+         var curveToRemove = Chart.Curves.Where(c => Equals(c.yData.Repository, dataRepository)).ToList();
+         if (curveToRemove.IsEmpty())
+            return;
+
+         curveToRemove.ForEach(curve => Chart.RemoveCurve(curve.Id) );
```

Comment:
Our method is each

Reply:
Each not ForEach

Reply:
@msevestre the nhibernate was actually because of the .IsEmpty on the list. .ForEach comes from the ms list